packages/runtime/src/useCases/transport/publicRelationshipTemplateReferences/GetPublicRelationshipTemplateReferences.ts

```diff
@@ -0,0 +1,20 @@
+import { Result } from "@js-soft/ts-utils";
+import { PublicRelationshipTemplateReferencesController } from "@nmshd/transport";
+import { Inject } from "@nmshd/typescript-ioc";
+import { PublicRelationshipTemplateReferenceDTO } from "../../../types/transport/PublicRelationshipTemplateReferenceDTO";
+import { UseCase } from "../../common";
+import { PublicRelationshipTemplateReferenceMapper } from "./PublicRelationshipTemplateReferenceMapper";
+
+export class GetPublicRelationshipTemplateReferencesUseCase extends UseCase<void, PublicRelationshipTemplateReferenceDTO[]> {
+    public constructor(@Inject private readonly publicRelationshipTemplateReferencesController: PublicRelationshipTemplateReferencesController) {
+        super();
+    }
+
+    protected async executeInternal(): Promise<Result<PublicRelationshipTemplateReferenceDTO[]>> {
+        const templateReferences = (await this.publicRelationshipTemplateReferencesController.getPublicRelationshipTemplateReferences()).map((reference) => {
+            return PublicRelationshipTemplateReferenceMapper.toPublicRelationshipTemplateReferenceDTO(reference);
+        });
```

Comment:
please move this to 2 calls, or do the mapping in Result ok - AND the mapper should have a function to transform a list of PublicRelationshipTemplateReferences